In [19]:
import pandas as pd
import numpy as np
from matplotlib import pyplot as plt
%matplotlib inline

## Ребров. Домашка. Урок 5




### Задание 1  
d_0 = 17mm  
a = 0.025  
n = 100  
std^2 = 4  
d_n = 17,5  
Тогда H0=d_0=17, H1=d_n>17  
Нахожу точку наблюдения и аргумент функции Лапласа:

In [20]:
x_ = 17.5
xn = 17
n = 100
std = 2
a = 0.025
u = (x_ - xn)*np.sqrt(n)/std
f_ = (1 - 2*a)/2
print(f'Точка наблюдения = {u}, аргумент ф-ии Лапласа = {f_}')

Точка наблюдения = 2.5, аргумент ф-ии Лапласа = 0.475


Аргументу 0.475  соответствует значение функции 1.96.  
Т.к. 1.96 < 2.5, гипотеза отвергается, и средний диаметр больше 17мм.

### Задание 2  
std = 16  
a = 0.05  
n = 256  
x_ = 80  
Мне показалось, что распределение Стьюдента для такой большой выборки не очень подходит, поэтому в качестве границ взял две сигмы.

In [21]:
n = 256
std = 16
D = std**2
t = 2
m = 80
print(f'Ответ. Доверительный интервал: {m - t*std/np.sqrt(n), m + t*std/np.sqrt(n)}')

Ответ. Доверительный интервал: (78.0, 82.0)


### Задание 3  
веса выборки: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190  
вес = 200  
a = 0.01
H0 = P = 200, H1 = p != 200  



In [24]:
samples = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
n = samples.shape[0]
x_ = samples.mean()
std = samples.std(ddof=1)
xn = 200
u = (x_ - xn)*np.sqrt(n)/std
f_ = (1 - 2*a)/2
print(f'Точка наблюдения = {u}, аргумент ф-ии Лапласа = {f_}')

198.5Точка наблюдения = -1.0651074037450896, аргумент ф-ии Лапласа = 0.475


Аргументу 0.475 соответствует значение функции 1.96  
Т.к 1.96 не равно 1.06, утверждение продавца не верно.